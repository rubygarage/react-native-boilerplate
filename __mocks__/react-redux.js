import * as reactRedux from 'react-redux';

export const dispatch = jest.fn();

export const useDispatch = jest.fn(() => dispatch);
export const useSelector = jest.fn();
export const { Provider } = reactRedux;
export const { connect } = reactRedux;
