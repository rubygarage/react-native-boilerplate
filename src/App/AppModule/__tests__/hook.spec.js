import { renderHook, act } from '@testing-library/react-hooks';
import { AppState } from 'react-native';

import {
  startListenConnectionState,
  stopListenConnectionState,
} from 'state/concepts/application/actions';
import { dispatch } from 'mocks/react-redux';

import useContainer from '../hook';

jest.spyOn(AppState, 'addEventListener');
jest.spyOn(AppState, 'removeEventListener');

describe('useContainer', () => {
  let result = null;
  let unmount = null;

  beforeEach(() => {
    ({ result, unmount } = renderHook(useContainer));
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  describe('Lifecycle', () => {
    it('checks mounting', () => {
      expect(dispatch).toHaveBeenCalledWith(startListenConnectionState());
      expect(AppState.addEventListener).toHaveBeenCalledWith('change', result.current.handleAppStateChange);
    });

    it('checks unmounting', () => {
      unmount();

      expect(AppState.removeEventListener).toHaveBeenCalledWith('change', result.current.handleAppStateChange);
    });
  });

  describe('checks handleAppStateChange method', () => {
    it('should start listen connection state', () => {
      act(() => {
        result.current.handleAppStateChange('active');
      });

      expect(dispatch).toHaveBeenCalledWith(startListenConnectionState());
    });

    it('should not start listen connection state', () => {
      act(() => {
        result.current.handleAppStateChange('background');
      });

      expect(dispatch).toHaveBeenCalledWith(stopListenConnectionState());
    });
  });
});
