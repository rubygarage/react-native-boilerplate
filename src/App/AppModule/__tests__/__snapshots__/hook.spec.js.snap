// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContainer matches snapshot 1`] = `
Object {
  "handleAppStateChange": [Function],
}
`;
